src/User/Module.php

```diff
@@ -56,6 +58,21 @@ class Module extends BaseModule
      * @var string prefix to be used as a replacement when user requeste deletion of his data.
      */
     public $GDPRanonymPrefix = 'GDPR';
+    /**
+     * @var bool if true, all registered users will be prompted to give consent if they have not gave not it earlier.
+     */
+    public $GDPRrequireConsentToAll = false;
```

Comment:
few days ago I was discussing with a friend about GDPR. He told me that theoretically you should ask for privacy policy acceptance **every time the policy changes**. So, if I registered to the site 1y ago and agreed with the PP, if the document changes I must agree again!

So I was wondering if, instead of storing a boolean value, we could store the date of the privacy policy date, and if the user accepted it before that date then we must require another consent.

What do you think?

Reply:
I started to wonder about the same thing yesterday. If you change the policy, how you can show that the user has agreed with that. Storing a timestamp instead of boolean seems a better to me. Ideally I would like to be able to trace back to the policy text. How about storing a hash of the policy so that one could verify the policy text also?

Reply:
I think storing the policy document is beyond the scope of the extension. If you know an user has accepted the policy on a certain date you should be able to assume the policy valid at that time was the one.

Reply:
I agree with all the above statements.